---
title: "Lab9"
author: "Karen Guerrero"
format: pdf
---


```{r}
PDB_data <- "Data_Export_Summary.csv"
PDB_data
```

```{r}
PDB <- read.csv(PDB_data)
PDB
```

```{r}
PDB$Multiple.methods <- as.numeric(gsub(",", "", PDB$Multiple.methods))
PDB$Neutron <- as.numeric(gsub(",", "", PDB$Neutron))
PDB$X.ray <- as.numeric(gsub(",", "", PDB$X.ray))
PDB$NMR <- as.numeric(gsub(",", "", PDB$NMR))
PDB$EM <- as.numeric(gsub(",", "", PDB$EM))
PDB$Total <- as.numeric(gsub(",", "", PDB$Total))
```

```{r}
(sum(PDB$X.ray)/sum(PDB$Total))*100
(sum(PDB$EM)/sum(PDB$Total))*100
```

##Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy?
The percentage of structures in the PDB that are solved by X-ray is 86.26097% and the percentage of structures solved by Electron Microscopy is 6.552983%. 


##Q2: What proportion of structures in the PDB are protein?

The proportion of structures in the PDB that are protein is 0.8698948. 

```{r}
PDB$Total[1]/sum(PDB$Total)
```



##Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

There are 4,707 protease structures in the current PDB. 


#Viewing PDB structures with Molstart

![](1HSG.png)


# Reading and working with structures in R
The 'bio3d' package for structural bioinformatics has lot's of features for reading and working with biomolecular sequences and structures. 

```{r}
library(bio3d)
pdb <- read.pdb("1hsg")
```


```{r}
pdb
```

##Q7: How many amino acid residues are there in this pdb object? 
There are 198 amino acid residues in this pdb object. 

##Q8: Name one of the two non-protein residues?
HOH is the name of one of the two non-protein residues. 

##Q9: How many protein chains are in this structure?
There are two protein chains in this structure. 


```{r}
attributes(pdb)
```

```{r}
head(pdb$atom)
```
```{r}
adk <- read.pdb("6s36")
```

```{r}
adk
```


Normal mode analysis (NMA) it is a bioinformatics method for prediciting functional motions. It will show us the parts of the protein that are "flexible" (i.e. most dynamic)

```{r}
# Perform flexiblity prediction
m <- nma(adk)
```

```{r}
plot(m)
```

```{r}
mktrj(m, file="adk_m7.pdb")
```




```{r}
## Install packages in the R console NOT your Rmd/Quarto file

# install.packages("bio3d")
# install.packages("devtools")
# install.packages("BiocManager")

# BiocManager::install("msa")
# devtools::install_bitbucket("Grantlab/bio3d-view")
```

##Q10. Which of the packages above is found only on BioConductor and not CRAN?

The packages only found on BioConductor and not CRAN is msa.

##Q11. Which of the above packages is not found on BioConductor or CRAN?
The package that is not found on BioConductor or CRAN is bio3d-view. 

##Q12. True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket?
True


#Compartive analysis of all ADK structures
First we get the sequence of ADK and use this to search the PDB database. 

```{r}
library(bio3d)
```

```{r}
aa <- get.seq("1ake_A")
```

```{r}
aa
```

##Q13. How many amino acids are in this sequence, i.e. how long is this sequence?
There are 214 amino acids in this sequence. 


```{r}
#Blast or hmmer search
b <- blast.pdb(aa)
```

```{r}
# Plot a summary of search results
```


```{r}
hits <- plot(b)
```


```{r}
# List out some 'top hits'
head(hits$pdb.id)
```

```{r}
hits <- NULL
hits$pdb.id <- c('1AKE_A','6S36_A','6RZE_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','3GMT_A','4PZL_A')
```


```{r}
# Download releated PDB files
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```

Viewing all these structures looks like a HOT mess!
We need to try something else...

We will align and supperpose these structures.

```{r}
# Align releated PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
pdbs
```


```{r}
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment (RStudio stated that image was too large when file was Render)
#plot(pdbs, labels=ids)
```


```{r}
anno <- pdb.annotate(ids)
unique(anno$source)
```

```{r}
anno
```

```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)
```

```{r}
# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```

```{r}
# Visualize first principal component
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")
```


```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(PC1=pc.xray$z[,1], 
                 PC2=pc.xray$z[,2], 
                 col=as.factor(grps.rd),
                 ids=ids)

p <- ggplot(df) + 
  aes(PC1, PC2, col=col, label=ids) +
  geom_point(size=2) +
  geom_text_repel(max.overlaps = 20) +
  theme(legend.position = "none")
p
```

```{r}
# NMA of all structures
modes <- nma(pdbs)
```

```{r}
plot(modes, pdbs, col=grps.rd)
```

##Q14. What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why? 

For the most part the green and pink lines fluctuations are similar but the black line's fluctuation is not similar to the colored lines. The black lines are different from the color lines. 






