---
title: "Lab11"
format: pdf
---

#Section 1. Proportion og G/G in a population

Downloaded csv file from Ensemble <https://uswest.ensembl.org/Homo_sapiens/Variation/Sample?db=core;r=17:39894595-39895595;v=rs8067378;vdb=variation;vf=105535077>

Here we read this csv file 

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```
```{r}
table(mxl$Genotype..forward.strand.)
```

```{r}
table(mxl$Genotype..forward.strand.) / nrow(mxl) *100
```


Let's look at a different population (GBR)

```{r}
gbr <- read.csv("373522-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(gbr)
```
Find proportion of G|G

```{r}
table(gbr$Genotype..forward.strand.)
```

```{r}
round(table(gbr$Genotype..forward.strand.) / nrow(gbr) *100 , 2)
```


This variant that is associated with childhood asthma is more frequent in the GBR population than the MXL population. 


## Section 4: Population Scale Analysis

One sample is obviously not enough to know what is happening in a population. You are
interested in assessing genetic differences on a population scale. So, you processed about
~230 samples and did the normalization on a genome level. Now, you want to find whether
there is any association of the 4 asthma-associated SNPs (rs8067378...) on ORMDL3
expression.
 
How many samples do we have?

```{r}
#download file from our classroom page
#rename to expr
expr <- read.table("rs8067378_ENSG00000172057.6.txt")
#View the first 6 rows of the file named expr
head(expr)
```

```{r}
#summary() tells us more information about the data imported
summary(expr)
```

```{r}
#Subsets all the rows that are A/A genotype
expr$geno == "A/A"
##Subsets in column exp by all the rows that are A/A genotype
expr$exp[expr$geno == "A/A"]
```

```{r}
#Obtains the median of A/A genotype
median(expr$exp[expr$geno == "A/A"])
```

```{r}
expr$geno == "A/G"
expr$exp[expr$geno == "A/G"]
```

```{r}
median(expr$exp[expr$geno == "A/G"])
```

```{r}
expr$geno == "G/G"
expr$exp[expr$geno == "G/G"]
```

```{r}
median(expr$exp[expr$geno == "G/G"])
```

##Q13: Read this file into R and determine the sample size for each genotype and their corresponding median expression levels for each of these genotypes.

There are 108 samples with the A/A genotype. There are 233 samples with the A/G genotype. There are 121 samples with the G/G genotype. 

A/A: 31.24847
A/G: 25.06486
G/G: 20.07363

```{r}
median(expr$exp[expr$geno == "A/A"])
median(expr$exp[expr$geno == "A/G"])
median(expr$exp[expr$geno == "G/G"])
```

```{r}
#Function used to determine sample size per genotype. 
table(expr$geno)
```


```{r}
#What to know how many sample are in the dataframe? This function will tell us the number of rows aka our samples.
nrow(expr)
```


```{r}
#Specifically tells us how many samples are found in each type:
table(expr$geno)
```

#Install ggplot2 so we could make a boxplot 

```{r}
library(ggplot2)
```

##Q14: Generate a boxplot with a box per genotype, what could you infer from the relative expression value between A/A and G/G displayed in this plot? Does the SNP effect the expression of ORMDL3?

Genotype A/A had higher levels of expression than genotype G/G. According to the boxplot the SNP does effect the expression of ORMDL3. 

##Following code is used to make ggplot 

```{r}
#input the dataframe first
#Aes(include x, y data, fill(is to indicate to fill in the box with the color)=color based off this type of data)
#geom_boxplot: function indicates to make a boxplot 
ggplot(expr) + aes(x=geno, exp, fill=geno) + geom_boxplot(notch=TRUE)
```



















